### PR TITLE
TM_1_1 idb test

### PR DESCRIPTION
**Pull request by nicHoch**

### CODE REVIEW

File: stixcore/tmtc/packet_factory.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        packet = self._check_registered(sph, idb)
+        return self.tm_packet_factory(packet, idb)
+
+
+class TMPacketFactory(BaseFactory):
+    """
+    Factory from TM packet return the correct type of based on the packet data and registered.
+    """
+    def __call__(self, data, idb):
+        return self._check_registered(data, idb)
```
Comment (samaloney) [inline stixcore/tmtc/packet_factory.py:93]: We'll have to think about this - when and how often should the IDB be queried. In theory every packet could need a different version of the IDB so maybe we need to create and instance of the `IDBManager` and and for each packet check if the current version is applicable or not?

File: stixcore/idb/idb.py [hunk 2]
```diff
@@ -572,3 +995,142 @@ def get_idb_version(self):
...
+        args = (service_type, service_subtype)
+        if ssid is not None:
+            args = args + (ssid,)
+        param_pcf_structures = self._execute(sql, args, 'dict')
+
+        repeater = [{'node': IdbPacketTree(), 'counter': 1024}]
+
+        for par in param_pcf_structures:
+            parObj = IdbVariableParameter(par)
+            if repeater:
```
Comment (samaloney) [inline stixcore/idb/idb.py:1122]: with `repeater` defined as above is this `if` not always true?

File: stixcore/tmtc/parser.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+from stixcore.util.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class PacketData:
+    """Generic class for organizing all parameters of the TM packet data."""
+
+
+def parameter_dict_2_PacketData(d):
```
Comment (samaloney) [inline stixcore/tmtc/parser.py:14]: Maybe make this a class method on the `PacketData` class?

**Review by samaloney — COMMENTED** [review]
Looks good, the reason some of the functions/classes are not showing up in the docs is they have not been added to the `__all__ = `  at the top of the relevant file.

File: stixcore/idb/idb.py [hunk 4]
```diff
@@ -11,11 +11,436 @@
 logger = get_logger(__name__)
 
 
+class IdbData:
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/idb.py:14]: good point i added the new super class

File: stixcore/idb/idb.py (hunk 2 above)
Comment (nicHoch) [reply stixcore/idb/idb.py:1122]: repeater.pop() can empty the repeater list and so this statement will become false

File: stixcore/tmtc/parser.py (hunk 3 above)
Comment (nicHoch) [reply stixcore/tmtc/parser.py:14]: good point

File: stixcore/tmtc/packet_factory.py (hunk 1 above)
Comment (nicHoch) [reply stixcore/tmtc/packet_factory.py:93]: made an issue for that

File: stixcore/idb/manager.py [hunk 5]
```diff
@@ -26,12 +25,11 @@
 logger = get_logger(__name__)
 
 
-class IDBManager:
-    """Manages IDB (definition of TM/TC packet structures) Versions
-    and provides a IDB reader
-    """
+class IdbManager:
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/manager.py:28]: grr i renamed that yesterday as i thought you suggested IdbManager.
Will change again

**Review by samaloney — APPROVED** [review]

**Pull request merged**